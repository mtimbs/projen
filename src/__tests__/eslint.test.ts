import { Eslint, NodeProject } from '..';
import { LogLevel } from '../logger';
import { mkdtemp, synthSnapshot } from './util';

test('devdirs', () => {
  // GIVEN
  const project = new NodeProject({
    outdir: mkdtemp(),
    name: 'test',
    logging: { level: LogLevel.OFF },
    defaultReleaseBranch: 'master',
  });

  // WHEN
  new Eslint(project, {
    devdirs: ['foo', 'bar'],
    dirs: ['mysrc'],
  });

  // THEN
  expect(synthSnapshot(project)['.eslintrc.json']).toMatchSnapshot();
});

describe('prettier', () => {
  test('snapshot', () => {
    // GIVEN
    const project = new NodeProject({
      outdir: mkdtemp(),
      name: 'test',
      logging: { level: LogLevel.OFF },
      defaultReleaseBranch: 'master',
    });

    // WHEN
    new Eslint(project, {
      dirs: ['mysrc'],
      prettier: true,
    });

    // THEN
    expect(synthSnapshot(project)['.eslintrc.json']).toMatchSnapshot();
  });

  test('error on formatting when enabled', () => {
    // GIVEN
    const project = new NodeProject({
      outdir: mkdtemp(),
      name: 'test',
      logging: { level: LogLevel.OFF },
      defaultReleaseBranch: 'master',
    });

    // WHEN
    const eslint = new Eslint(project, {
      dirs: ['mysrc'],
      prettier: true,
    });

    // THEN
    expect(eslint.rules).toHaveProperty('prettier/prettier', ['error']);
  });
});

describe('alias', () => {
  test('snapshot', () => {
    // GIVEN
    const project = new NodeProject({
      outdir: mkdtemp(),
      name: 'test',
      logging: { level: LogLevel.OFF },
      defaultReleaseBranch: 'master',
    });


    // WHEN
    new Eslint(project, {
      dirs: ['src'],

    });

    // THEN
    expect(synthSnapshot(project)['.eslintrc.json']).toMatchSnapshot();
  });

  test('custom config', () => {
    // GIVEN
    const project = new NodeProject({
      outdir: mkdtemp(),
      name: 'test',
      logging: { level: LogLevel.OFF },
      defaultReleaseBranch: 'master',
    });


    // WHEN
    const eslint = new Eslint(project, {
      dirs: ['src'],
      aliasMap: [
        ['@src', './src'],
      ],
      aliasExtensions: ['.ts', '.js'],
    });

    // THEN
    expect(eslint.config.settings['import/resolver'].alias).toHaveProperty('map', [['@src', './src']]);
    expect(eslint.config.settings['import/resolver'].alias).toHaveProperty('extensions', ['.ts', '.js']);
  });

});
