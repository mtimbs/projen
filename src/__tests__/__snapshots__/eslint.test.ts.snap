// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`alias snapshot 1`] = `
Object {
  "env": Object {
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "plugin:import/typescript",
  ],
  "ignorePatterns": Array [
    "*.js",
    "!.projenrc.js",
    "*.d.ts",
    "node_modules/",
    "*.generated.ts",
    "coverage",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        ".projenrc.js",
      ],
      "rules": Object {
        "@typescript-eslint/no-require-imports": "off",
        "import/no-extraneous-dependencies": "off",
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaVersion": 2018,
    "project": "./tsconfig.json",
    "sourceType": "module",
  },
  "plugins": Array [
    "@typescript-eslint",
    "import",
  ],
  "root": true,
  "rules": Object {
    "@typescript-eslint/indent": Array [
      "error",
      2,
    ],
    "@typescript-eslint/member-delimiter-style": Array [
      "error",
    ],
    "@typescript-eslint/member-ordering": Array [
      "error",
      Object {
        "default": Array [
          "public-static-field",
          "public-static-method",
          "protected-static-field",
          "protected-static-method",
          "private-static-field",
          "private-static-method",
          "field",
          "constructor",
          "method",
        ],
      },
    ],
    "@typescript-eslint/no-floating-promises": Array [
      "error",
    ],
    "@typescript-eslint/no-require-imports": Array [
      "error",
    ],
    "@typescript-eslint/no-shadow": Array [
      "error",
    ],
    "@typescript-eslint/return-await": Array [
      "error",
    ],
    "array-bracket-newline": Array [
      "error",
      "consistent",
    ],
    "array-bracket-spacing": Array [
      "error",
      "never",
    ],
    "brace-style": Array [
      "error",
      "1tbs",
      Object {
        "allowSingleLine": true,
      },
    ],
    "comma-dangle": Array [
      "error",
      "always-multiline",
    ],
    "comma-spacing": Array [
      "error",
      Object {
        "after": true,
        "before": false,
      },
    ],
    "curly": Array [
      "error",
      "multi-line",
      "consistent",
    ],
    "dot-notation": Array [
      "error",
    ],
    "import/no-extraneous-dependencies": Array [
      "error",
      Object {
        "devDependencies": Array [],
        "optionalDependencies": false,
        "peerDependencies": true,
      },
    ],
    "import/no-unresolved": Array [
      "error",
    ],
    "import/order": Array [
      "warn",
      Object {
        "alphabetize": Object {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": Array [
          "builtin",
          "external",
        ],
      },
    ],
    "indent": Array [
      "off",
    ],
    "key-spacing": Array [
      "error",
    ],
    "keyword-spacing": Array [
      "error",
    ],
    "max-len": Array [
      "error",
      Object {
        "code": 150,
        "ignoreComments": true,
        "ignoreRegExpLiterals": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true,
        "ignoreUrls": true,
      },
    ],
    "no-bitwise": Array [
      "error",
    ],
    "no-duplicate-imports": Array [
      "error",
    ],
    "no-multi-spaces": Array [
      "error",
      Object {
        "ignoreEOLComments": false,
      },
    ],
    "no-multiple-empty-lines": Array [
      "error",
    ],
    "no-return-await": Array [
      "off",
    ],
    "no-shadow": Array [
      "off",
    ],
    "no-trailing-spaces": Array [
      "error",
    ],
    "object-curly-newline": Array [
      "error",
      Object {
        "consistent": true,
        "multiline": true,
      },
    ],
    "object-curly-spacing": Array [
      "error",
      "always",
    ],
    "object-property-newline": Array [
      "error",
      Object {
        "allowAllPropertiesOnSameLine": true,
      },
    ],
    "quote-props": Array [
      "error",
      "consistent-as-needed",
    ],
    "quotes": Array [
      "error",
      "single",
      Object {
        "avoidEscape": true,
      },
    ],
    "semi": Array [
      "error",
      "always",
    ],
    "space-before-blocks": Array [
      "error",
    ],
  },
  "settings": Object {
    "import/parsers": Object {
      "@typescript-eslint/parser": Array [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": Object {
      "node": Object {},
      "typescript": Object {
        "project": "./tsconfig.json",
      },
    },
  },
}
`;

exports[`devdirs 1`] = `
Object {
  "env": Object {
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "plugin:import/typescript",
  ],
  "ignorePatterns": Array [
    "*.js",
    "!.projenrc.js",
    "*.d.ts",
    "node_modules/",
    "*.generated.ts",
    "coverage",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        ".projenrc.js",
      ],
      "rules": Object {
        "@typescript-eslint/no-require-imports": "off",
        "import/no-extraneous-dependencies": "off",
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaVersion": 2018,
    "project": "./tsconfig.json",
    "sourceType": "module",
  },
  "plugins": Array [
    "@typescript-eslint",
    "import",
  ],
  "root": true,
  "rules": Object {
    "@typescript-eslint/indent": Array [
      "error",
      2,
    ],
    "@typescript-eslint/member-delimiter-style": Array [
      "error",
    ],
    "@typescript-eslint/member-ordering": Array [
      "error",
      Object {
        "default": Array [
          "public-static-field",
          "public-static-method",
          "protected-static-field",
          "protected-static-method",
          "private-static-field",
          "private-static-method",
          "field",
          "constructor",
          "method",
        ],
      },
    ],
    "@typescript-eslint/no-floating-promises": Array [
      "error",
    ],
    "@typescript-eslint/no-require-imports": Array [
      "error",
    ],
    "@typescript-eslint/no-shadow": Array [
      "error",
    ],
    "@typescript-eslint/return-await": Array [
      "error",
    ],
    "array-bracket-newline": Array [
      "error",
      "consistent",
    ],
    "array-bracket-spacing": Array [
      "error",
      "never",
    ],
    "brace-style": Array [
      "error",
      "1tbs",
      Object {
        "allowSingleLine": true,
      },
    ],
    "comma-dangle": Array [
      "error",
      "always-multiline",
    ],
    "comma-spacing": Array [
      "error",
      Object {
        "after": true,
        "before": false,
      },
    ],
    "curly": Array [
      "error",
      "multi-line",
      "consistent",
    ],
    "dot-notation": Array [
      "error",
    ],
    "import/no-extraneous-dependencies": Array [
      "error",
      Object {
        "devDependencies": Array [
          "**/foo/**",
          "**/bar/**",
        ],
        "optionalDependencies": false,
        "peerDependencies": true,
      },
    ],
    "import/no-unresolved": Array [
      "error",
    ],
    "import/order": Array [
      "warn",
      Object {
        "alphabetize": Object {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": Array [
          "builtin",
          "external",
        ],
      },
    ],
    "indent": Array [
      "off",
    ],
    "key-spacing": Array [
      "error",
    ],
    "keyword-spacing": Array [
      "error",
    ],
    "max-len": Array [
      "error",
      Object {
        "code": 150,
        "ignoreComments": true,
        "ignoreRegExpLiterals": true,
        "ignoreStrings": true,
        "ignoreTemplateLiterals": true,
        "ignoreUrls": true,
      },
    ],
    "no-bitwise": Array [
      "error",
    ],
    "no-duplicate-imports": Array [
      "error",
    ],
    "no-multi-spaces": Array [
      "error",
      Object {
        "ignoreEOLComments": false,
      },
    ],
    "no-multiple-empty-lines": Array [
      "error",
    ],
    "no-return-await": Array [
      "off",
    ],
    "no-shadow": Array [
      "off",
    ],
    "no-trailing-spaces": Array [
      "error",
    ],
    "object-curly-newline": Array [
      "error",
      Object {
        "consistent": true,
        "multiline": true,
      },
    ],
    "object-curly-spacing": Array [
      "error",
      "always",
    ],
    "object-property-newline": Array [
      "error",
      Object {
        "allowAllPropertiesOnSameLine": true,
      },
    ],
    "quote-props": Array [
      "error",
      "consistent-as-needed",
    ],
    "quotes": Array [
      "error",
      "single",
      Object {
        "avoidEscape": true,
      },
    ],
    "semi": Array [
      "error",
      "always",
    ],
    "space-before-blocks": Array [
      "error",
    ],
  },
  "settings": Object {
    "import/parsers": Object {
      "@typescript-eslint/parser": Array [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": Object {
      "node": Object {},
      "typescript": Object {
        "project": "./tsconfig.json",
      },
    },
  },
}
`;

exports[`prettier snapshot 1`] = `
Object {
  "env": Object {
    "jest": true,
    "node": true,
  },
  "extends": Array [
    "plugin:import/typescript",
    "prettier",
    "plugin:prettier/recommended",
  ],
  "ignorePatterns": Array [
    "*.js",
    "!.projenrc.js",
    "*.d.ts",
    "node_modules/",
    "*.generated.ts",
    "coverage",
  ],
  "overrides": Array [
    Object {
      "files": Array [
        ".projenrc.js",
      ],
      "rules": Object {
        "@typescript-eslint/no-require-imports": "off",
        "import/no-extraneous-dependencies": "off",
      },
    },
  ],
  "parser": "@typescript-eslint/parser",
  "parserOptions": Object {
    "ecmaVersion": 2018,
    "project": "./tsconfig.json",
    "sourceType": "module",
  },
  "plugins": Array [
    "@typescript-eslint",
    "import",
    "prettier",
  ],
  "root": true,
  "rules": Object {
    "@typescript-eslint/member-ordering": Array [
      "error",
      Object {
        "default": Array [
          "public-static-field",
          "public-static-method",
          "protected-static-field",
          "protected-static-method",
          "private-static-field",
          "private-static-method",
          "field",
          "constructor",
          "method",
        ],
      },
    ],
    "@typescript-eslint/no-floating-promises": Array [
      "error",
    ],
    "@typescript-eslint/no-require-imports": Array [
      "error",
    ],
    "@typescript-eslint/no-shadow": Array [
      "error",
    ],
    "@typescript-eslint/return-await": Array [
      "error",
    ],
    "dot-notation": Array [
      "error",
    ],
    "import/no-extraneous-dependencies": Array [
      "error",
      Object {
        "devDependencies": Array [],
        "optionalDependencies": false,
        "peerDependencies": true,
      },
    ],
    "import/no-unresolved": Array [
      "error",
    ],
    "import/order": Array [
      "warn",
      Object {
        "alphabetize": Object {
          "caseInsensitive": true,
          "order": "asc",
        },
        "groups": Array [
          "builtin",
          "external",
        ],
      },
    ],
    "key-spacing": Array [
      "error",
    ],
    "no-bitwise": Array [
      "error",
    ],
    "no-duplicate-imports": Array [
      "error",
    ],
    "no-multiple-empty-lines": Array [
      "error",
    ],
    "no-return-await": Array [
      "off",
    ],
    "no-shadow": Array [
      "off",
    ],
    "no-trailing-spaces": Array [
      "error",
    ],
    "prettier/prettier": Array [
      "error",
    ],
  },
  "settings": Object {
    "import/parsers": Object {
      "@typescript-eslint/parser": Array [
        ".ts",
        ".tsx",
      ],
    },
    "import/resolver": Object {
      "node": Object {},
      "typescript": Object {
        "project": "./tsconfig.json",
      },
    },
  },
}
`;
